crates/paimon/src/spec/schema.rs

```diff
@@ -51,3 +51,138 @@ pub struct DataField {
     typ: DataType,
     description: Option<String>,
 }
+
+impl DataField {
+    pub fn new(id: i32, name: String, typ: DataType) -> Self {
+        Self {
+            id,
+            name,
+            typ,
+            description: None,
+        }
+    }
+
+    pub fn id(&self) -> i32 {
+        self.id
+    }
+
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn data_type(&self) -> &DataType {
+        &self.typ
+    }
+
+    pub fn description(&self) -> Option<&String> {
+        self.description.as_ref()
+    }
+
+    pub fn with_id(&self, new_id: i32) -> Self {
```

Comment:
Hi, `with_id` better to take `Self`. So users can use them like:

```rust
let df = DataField::new(id, name, typ)
    .with_id(new_id)
    .with_description(Some(description));
```

In this way, we can avoid unneeded allocation.

Reply:
Are you suggesting that we use mut self directly in the with method to update the contents of self, rather than calling the new method?

Reply:
done :)